src/components/accordion/use-div-height.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useRef, useState } from 'react';
+
+/**
+ * Lovely height measurement, from here:
+ * https://github.com/raunofreiberg/reach-ui/blob/6742f0cdcef05ba23077f63c21e3ede3bbfcae52/packages/accordion/examples/animated.example.js#L19-L39
+ */
+export function useDivHeight() {
+  const ref = useRef(null);
+  const [height, setHeight] = useState(0);
+
+  useEffect(() => {
+    const resizeObserver = new ResizeObserver(([entry]) => {
```

Comment:
Finally got how it works: the `ref` return variable is used as input for the function. I think this is a bit unintuitive, as I would expect the input to be passed as a function argument. 

I checked a popular hooks library how it's done there with similar use cases.  Check this code here how they use the `useIntersection `:
https://github.com/streamich/react-use/blob/master/docs/useIntersection.md 

They put the `ref` outside of the function so that you can pass it in as a function argument.

```js
const intersectionRef = React.useRef(null);
const intersection = useIntersection(interestectionRef)
```

I think this is simpler (or more intuitive) to read and would therefore recommend to put the `ref` outside.

Reply:
Thanks for the link to useIntersection. Makes sense to make this more flexible. I copied this part from the linked sources without thinking too much over details like that. I very much appreciate that specific point though and I'm going to refactor the code.